core/components/tree/basenode.js

```diff
@@ -0,0 +1,1151 @@
+      if (this.isRightToLeft()) {
+        row.style.paddingRight = indent;
+      } else {
+        row.style.paddingLeft = indent;
+      }
+    }
+    this.forEachChild(function(child) { child.setDepth_(depth + 1); });
+  }
+};
+
+/**
+ * Returns true if the node is a descendant of this node
+ * @param {Blockly.tree.BaseNode} node The node to check.
+ * @return {boolean} True if the node is a descendant of this node, false
+ *    otherwise.
+ */
+Blockly.tree.BaseNode.prototype.contains = function(node) {
+  var current = node;
+  while (current) {
+    if (current == this) {
+      return true;
+    }
+    current = current.getParent();
+  }
+  return false;
+};
+
+/**
+ * An array of empty children to return for nodes that have no children.
+ * @type {!Array<!Blockly.tree.BaseNode>}
+ * @private
+ */
+Blockly.tree.BaseNode.EMPTY_CHILDREN_ = [];
+
+/**
+ * @param {number} index 0-based index.
+ * @return {Blockly.tree.BaseNode} The child at the given index; null if none.
+ */
+Blockly.tree.BaseNode.prototype.getChildAt;
+
+/**
+ * Returns the children of this node.
+ * @return {!Array<!Blockly.tree.BaseNode>} The children.
+ */
+Blockly.tree.BaseNode.prototype.getChildren = function() {
+  var children = [];
+  this.forEachChild(function(child) { children.push(child); });
+  return children;
+};
+
+/**
+ * @return {Blockly.tree.BaseNode} The first child of this node.
+ */
+Blockly.tree.BaseNode.prototype.getFirstChild = function() {
+  return this.getChildAt(0);
+};
+
+/**
+ * @return {Blockly.tree.BaseNode} The last child of this node.
+ */
+Blockly.tree.BaseNode.prototype.getLastChild = function() {
+  return this.getChildAt(this.getChildCount() - 1);
+};
+
+/**
+ * @return {Blockly.tree.BaseNode} The previous sibling of this node.
+ */
+Blockly.tree.BaseNode.prototype.getPreviousSibling = function() {
+  return this.previousSibling_;
+};
+
+/**
+ * @return {Blockly.tree.BaseNode} The next sibling of this node.
+ */
+Blockly.tree.BaseNode.prototype.getNextSibling = function() {
+  return this.nextSibling_;
+};
+
+/**
+ * @return {boolean} Whether the node is the last sibling.
+ */
+Blockly.tree.BaseNode.prototype.isLastSibling = function() {
+  return !this.nextSibling_;
+};
+
+/**
+ * @return {boolean} Whether the node is selected.
+ */
+Blockly.tree.BaseNode.prototype.isSelected = function() {
+  return this.selected_;
+};
+/**
+ * Selects the node.
+ */
+Blockly.tree.BaseNode.prototype.select = function() {
+  var tree = this.getTree();
+  if (tree) {
+    tree.setSelectedItem(this);
+  }
+};
+
+/**
+ * Called from the tree to instruct the node change its selection state.
+ * @param {boolean} selected The new selection state.
+ * @protected
+ */
+Blockly.tree.BaseNode.prototype.setSelectedInternal = function(selected) {
+  if (this.selected_ == selected) {
+    return;
+  }
+  this.selected_ = selected;
+
+  this.updateRow();
+
+  var el = this.getElement();
+  if (el) {
+    Blockly.utils.aria.setState(el, 'selected', selected);
```

Comment:
I haven't looked at the utils.aria stuff.  Do you also need to unset selected and activedescendant on the previously selected element.

Reply:
Yep, when a new item is selected, the currently selected item gets a setSelected(false) before the new one gets setSelected(true). 
Login in Blockly.tree.TreeControl.prototype.setSelectedItem